pkg/bgpv1/gobgp/conversions.go

```diff
@@ -222,3 +235,52 @@ func toGoBGPSoftResetDirection(direction types.SoftResetDirection) gobgp.ResetPe
 	}
 	return gobgp.ResetPeerRequest_BOTH
 }
+
+func toGoBGPExtendedCommunity(ec types.ExtendedCommunity) string {
+	var prefix string
+	switch ec.SubType {
+	case types.ExtendedCommunityRouteTarget:
+		prefix = routeTargetPrefix
+	case types.ExtendedCommunityRouteOrigin:
+		prefix = routeOriginPrefix
+	case types.ExtendedCommunityLinkBandwidth:
+		prefix = linkBandwidthPrefix
+	}
+	return prefix + ":" + ec.Value
+}
+
+func toAgentExtendedCommunity(ec string) (res types.ExtendedCommunity, err error) {
+	parts := strings.SplitN(ec, ":", 2)
+	if len(parts) < 2 {
+		return res, fmt.Errorf("invalid format of extended commnity (%s)", ec)
+	}
+	switch parts[0] {
+	case routeTargetPrefix:
+		res.SubType = types.ExtendedCommunityRouteTarget
+	case routeOriginPrefix:
+		res.SubType = types.ExtendedCommunityRouteOrigin
+	case linkBandwidthPrefix:
+		res.SubType = types.ExtendedCommunityLinkBandwidth
+	}
+	res.Value = parts[1]
+	return res, nil
+}
+
+func toGoBGPExtendedCommunities(ecs []types.ExtendedCommunity) []string {
+	res := make([]string, 0, len(ecs))
+	for _, ec := range ecs {
+		res = append(res, toGoBGPExtendedCommunity(ec))
+	}
+	return res
+}
+
+func toAgentExtendedCommunities(ecs []string) []types.ExtendedCommunity {
+	res := make([]types.ExtendedCommunity, 0, len(ecs))
+	for _, ec := range ecs {
+		community, err := toAgentExtendedCommunity(ec)
+		if err == nil {
+			res = append(res, community)
```

Comment:
should we log the error, if there is a deviation in the expected translation we will catch it here.